EIPS/eip-5218.md

```diff
@@ -0,0 +1,252 @@
+
+The full log of license creation, transfer, and revocation *must* be traceable via event logs. Therefore, all license creations and transfers *must* emit a corresponding log event. Revocation may differ a bit. An implementation of this EIP may emit a `Revoke` event only when a license is revoked in a function call, or for every revoked license, both are sufficient to trace the status of all licenses. The former costs less gas if revoking a license automatically revokes all sublicenses under it, while the latter is efficient in terms of interrogation of a license status. Implementers should make the tradeoffs depending on their license terms.
+
+The `revoker` of a license may be the licensor, the license holder, or a smart contract address which calls the `revokeLicense` function when some conditions are met. Implementers should be careful with the authorization, and may make the `revoker` smart contract forward compatible with transfers by not hardcoding the addresses of `licensor` or `licenseHolder`.
+
+The license `URI` may point to a JSON file that conforms to the "ERC5218 Metadata JSON Schema" as below, which adopts the ["three-layer"](https://creativecommons.org/licenses/) design of the Creative Commons Licenses:
+
+```json
+{
+    "title": "License Metadata",
+    "type": "object",
+    "properties": {
+        "legal-code": {
+            "type": "string",
+            "description": "The legal code of the license."
+        },
+        "human-readable": {
+            "type": "string",
+            "description": "The human readable license deed."
+        },
+        "machine-readable": {
+            "type": "string",
+            "description": "The machine readable code of the license that can be recognized by software, such as CC REL."
+        }
+    }
+}
+```
+
+The `supportsInterface` method MUST return `true` when called with `0xXXX(TODO)`.
+
+## Rationale
+
+This EIP aims to allow tracing all licenses to an NFT to facilitate right management. The ERC-721 standard only logs the property but not the legal rights tethered to NFTs. Even when logging the license via the optional ERC721Metadata extension, sublicenses are not traceable, which doesn't comply with the transparency goals of Web3. Some implementations attempt to get around this limitation by minting NFTs to represent a particular license, such as the [BAYC #6068 Royalty-Free Usage License](https://opensea.io/assets/0x495f947276749ce646f68ac8c248420045cb7b5e/111558001269395431138572590975405607139590663093731712964826342979501444572944). This is not an ideal solution because the linking between different licenses to an NFT is ambiguous. An auditor has to investigate all NFTs in the blockchain and inspect the metadata which hasn't been standardized in terms of sublicense relationship. To avoid these problems, this EIP logs all licenses to an NFT in a tree data structure, which is compatible with ERC721 and allows efficient traceability.
```

Comment:
```suggestion
This EIP aims to allow tracing all licenses to an NFT to facilitate right management. The ERC-721 standard only logs the property but not the legal rights tethered to NFTs. Even when logging the license via the optional ERC721Metadata extension, sublicenses are not traceable, which doesn't comply with the transparency goals of Web3. Some implementations attempt to get around this limitation by minting NFTs to represent a particular license, such as the BAYC #6068 Royalty-Free Usage License. This is not an ideal solution because the linking between different licenses to an NFT is ambiguous. An auditor has to investigate all NFTs in the blockchain and inspect the metadata which hasn't been standardized in terms of sublicense relationship. To avoid these problems, this EIP logs all licenses to an NFT in a tree data structure, which is compatible with ERC721 and allows efficient traceability.
```